### DISCUSSION TITLE
How to use technical analysis tools on Deriv’s trading bot

### DISCUSSION THREAD

**Discussion by alanvito1** [alanvito1/ORSTAC#65] [General]
Using technical analysis on DBot works the same way as using it on any other trading platform – it helps predict the asset’s price movements to make informed trading decisions. But while on other platforms you analyse price chart patterns or apply [technical indicators](https://blog.deriv.com/posts/the-4-most-common-types-of-technical-indicators/) manually, with DBot, these processes are automated with pre-set blocks. 

If you are not familiar with the concept of technical analysis, we’ve covered it extensively in our [What is technical analysis in trading](https://blog.deriv.com/posts/what-is-technical-analysis-in-trading/) blog. To put it briefly, technical analysis is a process of analysing past data of financial markets, such as price and trading volume, to predict future price movements.

Let’s go through the details of how it works on DBot.

Ticks and candles analysis
The simplest way of using technical analysis on DBot is to use the Tick and candle analysis blocks, which can be found under its respective subtab in the ‘Analysis’ tab.

 5.1. Tick and Candle Analysis Blocks on D Bot – Deriv's Trading Bot

These blocks analyse the prices of the previous ticks, candles, or the entire market direction.  You can find a more detailed description of the functionality of each block by clicking the ‘Learn more’ next to them.

Here is a simple example of how to set up a basic options trading strategy that will purchase a contract based on the last digit of the previous tick: 

Select the ‘[Conditional block](https://blog.deriv.com/posts/how-to-set-up-advanced-parameters-for-derivs-trading-bot/)‘ and drag it to your mandatory ‘[Purchase block](https://blog.deriv.com/posts/how-to-build-a-basic-trading-bot-with-dbot/)’.
Select the ‘Compare block’ and drag it into the empty placeholder of your ‘Conditional’ block.
Select the ‘Last digit’ block from the ‘Tick and candle analysis’ subtab of the ‘Analysis tab’, and drag it into the first placeholder of your ‘Compare’ block.
Select the ‘Number block’, drag it into the second placeholder of your ‘Compare’ block, and select the rule (equal, greater etc.)
Select the type of contract for your trade – rise or fall in our example.
Purchase Conditions Block on D Bot with Simple Tick and Candle Analysis Strategy

Once you complete all these steps, your ‘Purchase conditions’ block is ready. The conditions we’ve just set will instruct your trading bot to buy a Rise contract whenever the last digit of the previous tick is not equal to 4.

That this particular strategy is only an example of how you can use the Tick and candle analysis blocks, and doesn’t carry any value in a proper trading environment.

Technical indicators in DBot
Another way of using technical analysis with your trading bot is to apply technical indicator blocks that can be found in the Indicators subtab of the ‘Analysis’ tab.

5.2. Technical Indicators Blocks on D Bot – Deriv's Trading Bot

DBot has 4 main Indicators blocks to help you evaluate all the main signals that can predict future price movements:

 Simple moving average (SMA)
 Bollinger bands (BB)
Relative strength index (RSI)
Moving Average Convergence Divergence (MACD)
The other blocks in the Indicators subtab represent advanced variations of these 4 main blocks. But if you are just starting out, let’s focus on setting up the main indicators, taking Bollinger bands as an example.

In a price chart, the Bollinger bands indicator looks like a channel made of 3 lines. The simple rule is that whenever the price breaks out of one of the outer lines, it tends to move back to the middle line, serving as a buy signal for traders.

Bolinger Bands Technical Indicators on Deriv

With DBot, you don’t need to monitor the price chart to catch the moment when the breakout happens – you can just instruct your bot to do it and buy a contract when it happens:

1. Select the ‘Bollinger bands’ block from the list of indicators and drag it to your ‘Purchase conditions’ block.

2. Click the ‘bb’ variable, select the ‘Rename variable’ option from the dropdown, and assign the ‘bb down’ name to it.

3. Click the ‘middle’ variable, and select the ‘lower’ value from the dropdown.

4. Select the ‘Tick list’ block from ‘Tick and candle analysis’ and insert it in the placeholder next to the Input list.

5. The ‘Period’ block shows the number of past ticks the indicator will use to analyse the market trend, while the ‘Standard Deviation Up/Down Multiplier blocks show how wide the outer lines of the indicator are. For this example, let’s set it to 20, 2 and 2, respectively.

Purchase Conditions Block on D Bot with Simple Technical Indicators Strategy

To understand how these numbers are reflected visually, you can go to the ‘Chart’ tab at the bottom of the left corner of your workspace and apply the Bollinger bands indicator in the price chart too. Right-click inside the grey channel and change the numbers in the window that pops up and click ‘Done’.

Bolinger Bands Technical Indicators on D Bot – Deriv's Trading Bot

6. Select the ‘Conditional block’ and place it right before the ‘Purchase (Rise)’ block, add the ‘Compare’ block to it, just like we did when we set up a trading strategy based on the last digit in the previous example.

7. Select the ‘Last tick’ block from ‘Tick and candle analysis’ and drag it into the first placeholder of your ‘Compare’ block.

8. Go to your Variables tab and select the ‘bb down’ variable we created earlier, placing it into the second placeholder of your ‘Compare’ block.

9. Set the rule to Last tick is less than the ‘bb down’ so that your bot takes action when the previous tick is below the lower Bollinger band.

10. Set the ‘Purchase’ block to ‘Rise’.

Setting up A Trading Strategy on D Bot Using Technical Indicators

This strategy tells your trading bot to buy a Rise contract whenever a previous tick is lower than the bottom line of the Bollinger bands indicator.

Here is how it will look like in your DBot platform:

Purchase Conditions Block on Deriv's D Bot with Simple Technical Indicators Strategy

Taking this strategy as an example, you can add one more ‘Bollinger bands’ block, replicating the same actions but for the upper line of the indicator. You can also customise your strategy in different ways, for example, setting up a rule where the Last tick is not just less than ‘bb down’ but also equals it. 

Following the same methods, you can use other technical indicators on DBot to help your trading bot determine the best time to purchase a contract. 

This sums up a complete overview of setting up a trading strategy with DBot – from setting up basic and advanced parameters to teaching your bot how to analyse markets.

Now you can test your newly acquired knowledge on your risk-free [demo account,](https://app.deriv.com/bot) preloaded with a 10,000 USD of virtual currency, or check our bonus [Top 5 tips and tricks for your DBot trading strategy](https://blog.deriv.com/posts/top-5-tips-and-tricks-for-your-dbot-trading-strategy/) blog to find out a few more little hacks to make your DBot journey better.

**Comment by NoirWayne** [comment]
can u write this two examples in code ? i wanna put them to a test

**Comment by DKFXAUTO** [comment]
Do you have a telegram account I can message you directly?

**Comment by Nazehs** [comment]
I am wondering how i can use macd indicator to trade with dbot. I have tried using it and it seems not to be working for because i am not sure of which of the item that is actually my current macd/signal line since its an array by default
